lib/TopStoriesFeed.jsm

```diff
@@ -65,7 +67,10 @@ this.TopStoriesFeed = class TopStoriesFeed {
     // if needed lazy load activity stream top stories feed based on
     // actual user preference when INIT and PREF_CHANGED is invoked
     this.discoveryStreamEnabled =
-      ds && ds.value && JSON.parse(ds.value).enabled;
+      ds &&
+      ds.value &&
+      JSON.parse(ds.value).enabled &&
+      Services.prefs.getBoolPref(DISCOVERY_STREAM_PREF_ENABLED, false);
```

Comment:
@Mardak So this is defaulted to `false/undefined`.

What this means is as soon as we ship this patch as is, beta and nightly revert to AS.

You think it's worth doing a check here to see if we're beta/nightly/unbranded and still turning this on? But then we couldn't test that this pref turns things off.

Reply:
may be we should check here if [prefHasuserValue](https://searchfox.org/mozilla-central/source/modules/libpref/nsIPrefBranch.idl#) that is an explicit set value, if not we ignore

Reply:
Rollout uses default pref branch, so checking "user" prefs will not work.